multimodal/src/autogluon/multimodal/problem_types.py

```diff
@@ -0,0 +1,221 @@
+"""Problem types supported in MultiModalPredictor"""
+
+from dataclasses import dataclass, field
+from typing import Optional, Set
+
+from .constants import (
+    BINARY,
+    CATEGORICAL,
+    CLASSIFICATION,
+    FEATURE_EXTRACTION,
+    FEW_SHOT_TEXT_CLASSIFICATION,
+    IMAGE,
+    IMAGE_BYTEARRAY,
+    IMAGE_SIMILARITY,
+    IMAGE_TEXT_SIMILARITY,
+    MULTICLASS,
+    NAMED_ENTITY_RECOGNITION,
+    NER,
+    NER_ANNOTATION,
+    NUMERICAL,
+    OBJECT_DETECTION,
+    OCR_TEXT_DETECTION,
+    OCR_TEXT_RECOGNITION,
+    REGRESSION,
+    ROIS,
+    TEXT,
+    TEXT_NER,
+    TEXT_SIMILARITY,
+    ZERO_SHOT_IMAGE_CLASSIFICATION,
+)
+from .registry import Registry
+
+problem_type_reg = Registry("problem_types")
+
+
+@dataclass
+class ProblemType:
+    """Problem type class. Stores the name of the problem
+    and its related properties. Some of the properties will be useful for the label / feature inference logic."""
+
+    name: str  # Name of the problem
+    support_fit: bool = True  # Whether the problem type support `.fit()`
+    inference_ready: bool = False  # Support `.predict()` and `.evaluate()` without calling `.fit()`
```

Comment:
I think inference_ready is better to be a state of predictor.

Reply:
This indicates whether the problem supports zero-shot inference.

Reply:
How about using name `support_zero_shot`? I think we may need `inference_ready ` in predictor in later refactoring.

Reply:
Sounds good. support_zero_shot is reasonable.